### ISSUE TITLE
Fix S2292: FP on volatile fields

### ISSUE THREAD

**Issue by DRKV333** [SonarSource/sonar-dotnet#3107]
### Description

S2291 is triggered on trivial properties that use a backing field marked with `volatile` .  

### Repro steps

```
class VolatileBackingFieldTest
{
    public bool Flag //S2292 here
    {
        get => flag;
        set => flag = value;
    }

    private volatile bool flag;
}
```

### Expected behavior

Properties that use a volatile backing field should be exempt from S2292, since it is not possible to express this using an auto-implemented property.

### Actual behavior

S2292 is still triggered.

### Known workarounds

Manually suppress warning.

### Related information

* SonarC# Version 8.3.0.14607
* Visual Studio Community Version 2019 16.4.3
* SonarLint Extension Version 4.17.0.14702
